src/entities/activity/ui/items/ActionPlan/Document.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { forwardRef, useEffect, useState, useContext, useMemo, useCallback } from 'react';
+
+import { Box } from '@mui/material';
+import { v4 as uuidV4 } from 'uuid';
+
+import { DocumentContext } from './DocumentContext';
+import { useAvailableBodyWidth, usePageMaxHeight } from './hooks';
+import { Page } from './Page';
+import { extractActivitiesPhrasalData } from './phrasalData';
+
+import { getProgressId } from '~/abstract/lib';
+import { PhrasalTemplatePhrase } from '~/entities/activity/lib';
+import { appletModel } from '~/entities/applet';
+import { SurveyContext } from '~/features/PassSurvey';
+import { useAppSelector } from '~/shared/utils';
+import measureComponentHeight from '~/shared/utils/measureComponentHeight';
+
+type DocumentProps = {
+  phrases: PhrasalTemplatePhrase[];
+  appletTitle: string;
+  phrasalTemplateCardTitle: string;
+};
+
+type IdentifiblePhrasalTemplatePhrase = PhrasalTemplatePhrase & { id: string };
+
+export const Document = forwardRef<HTMLDivElement, DocumentProps>(
+  ({ appletTitle, phrases, phrasalTemplateCardTitle }, ref) => {
+    const context = useContext(SurveyContext);
+
+    const noImage = useMemo(
+      () => phrases.filter((phrase) => !!phrase.image).length <= 0,
+      [phrases],
+    );
+
+    const identifiblePhrases = useMemo(
+      () =>
+        phrases.map<IdentifiblePhrasalTemplatePhrase>((phrase) => {
+          return {
+            ...phrase,
+            id: uuidV4(),
+          };
+        }),
+      [phrases],
+    );
+
+    const activityProgress = useAppSelector((state) =>
+      appletModel.selectors.selectActivityProgress(
+        state,
+        getProgressId(context.activityId, context.eventId),
```

Comment:
`getProgressId` also needs to have the `targetSubjectId` passed to it from context for this screen not to crash when performing a multi-informant assignment:
```suggestion
        getProgressId(context.activityId, context.eventId, context.targetSubject?.id),
```